### PR TITLE
Implementing soft delete and error codes

### PR DESCRIPTION
**Pull request by vilanovi**
- Soft delete in RawSQLDataSource and SQLRowCounterDataSource
- New error code attribute (optional) in error classes

All changes should be retro compatible with previous versions of harmony-core

### CODE REVIEW

File: packages/core/src/repository/data-source/sql/raw-sql.data-source.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ import {
     IdQuery,
     IdsQuery,
     PaginationOffsetLimitQuery,
-    Query,
+    Query, BaseColumnDeletedAt,
```
Comment (doup) [inline packages/core/src/repository/data-source/sql/raw-sql.data-source.ts:18]: One import per line?

File: packages/core/src/repository/data-source/sql/raw-sql.data-source.ts [hunk 2]
```diff
@@ -71,6 +72,8 @@ export class RawSQLDataSource implements GetDataSource<RawSQLData>, PutDataSourc
         this.tableColumns = tableColumns.concat(columns);
     }
 
+    public softDeleteEnabled = false;
+
```
Comment (doup) [inline packages/core/src/repository/data-source/sql/raw-sql.data-source.ts:76]: I though you were going to pass this on the constructor or derive it from `deleteAtColumn`. Make `deleteAtColumn` optional without a default value and if it's defined the assume that it's soft delete?

File: packages/core/src/repository/errors.ts [hunk 3]
```diff
@@ -1,71 +1,83 @@
+
+export const UnknownErrorCode = -1;
+
+export const MethodNotImplementedErrorName =  'ethod Not Implemented';
```
Comment (doup) [inline packages/core/src/repository/errors.ts:4]: `M` is missing on the string. `ethod` => `Method`

File: packages/core/src/repository/data-source/sql/raw-sql.data-source.ts (hunk 1 above)
Comment (vilanovi) [reply packages/core/src/repository/data-source/sql/raw-sql.data-source.ts:18]: Does it matter?

File: packages/core/src/repository/data-source/sql/raw-sql.data-source.ts (hunk 2 above)
Comment (vilanovi) [reply packages/core/src/repository/data-source/sql/raw-sql.data-source.ts:76]: That's correct. I just rushed to code what I had before. Good catch.

File: packages/core/src/repository/errors.ts (hunk 3 above)
Comment (vilanovi) [reply packages/core/src/repository/errors.ts:4]: Excelent catch.

File: packages/core/src/repository/data-source/sql/raw-sql.data-source.ts (hunk 1 above)
Comment (doup) [reply packages/core/src/repository/data-source/sql/raw-sql.data-source.ts:18]: Just a matter of style, it's not relevant.

File: packages/core/src/repository/errors.ts (hunk 3 above)
Comment (vilanovi) [reply packages/core/src/repository/errors.ts]: Solved

File: packages/core/src/repository/data-source/sql/raw-sql.data-source.ts (hunk 2 above)
Comment (vilanovi) [reply packages/core/src/repository/data-source/sql/raw-sql.data-source.ts]: moved to the constructor.

**Pull request merged**